component/core/src/test/java/org/exoplatform/social/core/jpa/storage/ProfileSettingStorageTest.java

```diff
@@ -0,0 +1,142 @@
+ *
+ * Copyright (C) 2023 Meeds Association contact@meeds.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.exoplatform.social.core.jpa.storage;
+
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.social.core.jpa.test.AbstractCoreTest;
+import org.exoplatform.social.core.profileproperty.model.ProfilePropertySetting;
+import org.exoplatform.social.core.storage.api.ProfileSettingStorage;
+
+/**
+ * Unit Tests for {@link ProfileSettingStorage}
+ */
+public class ProfileSettingStorageTest extends AbstractCoreTest {
+
+  private final Log             LOG = ExoLogger.getLogger(ProfileSettingStorageTest.class);
+
+  private ProfileSettingStorage profileSettingStorage;
+
+  @Override
+  public void setUp() throws Exception {
+    super.setUp();
+    profileSettingStorage = getContainer().getComponentInstanceOfType(ProfileSettingStorage.class);
+    assertNotNull(profileSettingStorage);
+  }
+
+  public void testSaveProfilePropertySetting() {
+    try {
+      ProfilePropertySetting profilePropertySetting = createProfileSettingInstance("test");
+      profilePropertySetting = profileSettingStorage.saveProfilePropertySetting(profilePropertySetting, true);
+      profilePropertySetting = profileSettingStorage.findProfileSettingByName(profilePropertySetting.getPropertyName());
+      assertNotNull(profilePropertySetting.getId());
+    } catch (Exception e) {
+      LOG.error(e.getMessage());
+      fail(e.getMessage());
+    }
+  }
+
+  public void testDeleteProfilePropertySetting() {
+    try {
+      ProfilePropertySetting profilePropertySetting = createProfileSettingInstance("test");
+      profilePropertySetting = profileSettingStorage.saveProfilePropertySetting(profilePropertySetting, true);
+      profilePropertySetting = profileSettingStorage.findProfileSettingByName(profilePropertySetting.getPropertyName());
+      assertNotNull(profilePropertySetting.getId());
+      profileSettingStorage.deleteProfilePropertySetting(profilePropertySetting.getId());
+      profilePropertySetting = profileSettingStorage.findProfileSettingByName(profilePropertySetting.getPropertyName());
+      assertNull(profilePropertySetting);
```

Comment:
please test on other object attributes to make the test more relevant and increase more the coverage.